sktime/forecasting/base/adapters/_pmdarima.py

```diff
@@ -117,45 +131,162 @@ def _predict_in_sample(
-            )
-            return y_pred, pred_int
+            pred_ints = []
+            for a in alpha:
+                result = self._forecaster.predict(
+                    n_periods=n_periods,
+                    X=X,
+                    return_conf_int=True,
+                    alpha=a,
+                )
+                pred_int = result[1]
+                pred_int = pd.DataFrame(
+                    pred_int[fh_idx, :], index=fh_abs, columns=["lower", "upper"]
+                )
+                pred_ints.append(pred_int)
+            return result[0], pred_ints
         else:
             return pd.Series(result[fh_idx], index=fh_abs)
 
+    def _predict_quantiles(self, fh, X=None, alpha=DEFAULT_ALPHA):
+        """
+        Compute/return prediction quantiles for a forecast.
+
+        Must be run *after* the forecaster has been fitted.
+
+        If alpha is iterable, multiple quantiles will be calculated.
+
+        Parameters
+        ----------
+        fh : int, list, np.array or ForecastingHorizon
+            Forecasting horizon, default = y.index (in-sample forecast)
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series
+        alpha : float or list of float, optional (default=[0.05, 0.95])
+            A probability or list of, at which quantile forecasts are computed.
+
+        Returns
+        -------
+        quantiles : pd.DataFrame
+            Column has multi-index: first level is variable name from y in fit,
+                second level being the values of alpha passed to the function.
+            Row index is fh. Entries are quantile forecasts, for var in col index,
+                at quantile probability in second col index, for the row index.
+        """
+        self.check_is_fitted()
+        alpha = check_alpha(alpha)
+
+        # convert alpha to the one needed for predict intervals
+        alphas = []
+        for a in alpha:
+            if a < 0.5:
+                alphas.append(1 - ((0.5 - a) * 2))
+            elif a > 0.5:
+                alphas.append(1 - ((a - 0.5) * 2))
+            else:
+                alphas.append(0)
+
+        fh_oos = self.fh.to_out_of_sample(self.cutoff)
+        fh_ins = self.fh.to_in_sample(self.cutoff)
+
+        kwargs = {"X": X, "return_pred_int": True, "alpha": alphas}
+        # all values are out-of-sample
+        if self.fh.is_all_out_of_sample(self.cutoff):
+            y_oos_pred, y_oos_pred_int = self._predict_fixed_cutoff(fh_oos, **kwargs)
+            pred_quantiles = pd.DataFrame()
+            for oos_intervals, a in zip(y_oos_pred_int, alpha):
+                if a < 0.5:
+                    pred_quantiles[a] = oos_intervals["lower"]
+                else:
+                    pred_quantiles[a] = oos_intervals["upper"]
+            arrays = [len(alpha) * ["Quantiles"], alpha]
+            index = pd.MultiIndex.from_tuples(list(zip(*arrays)))
```

Comment:
if you use `from_prod`, no need to use `zip`

Reply:
edited